<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>UStatDecouple: Probabilistic Decoupling for U-Statistics in Genomic Analysis • UStatDecouple</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="UStatDecouple: Probabilistic Decoupling for U-Statistics in Genomic Analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">UStatDecouple</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/UStatDecouple.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/danymukesha/UStatDecouple/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>UStatDecouple: Probabilistic Decoupling for U-Statistics in Genomic Analysis</h1>
                        <h4 data-toc-skip class="author">Dany
Mukesha</h4>
            
            <h4 data-toc-skip class="date">2026-01-14</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/danymukesha/UStatDecouple/blob/main/vignettes/UStatDecouple.Rmd" class="external-link"><code>vignettes/UStatDecouple.Rmd</code></a></small>
      <div class="d-none name"><code>UStatDecouple.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Decoupling in probability involves replacing a dependent sum
      (like a U-statistic) with an independent version that is easier to
      bound. For example, if you have a statistic
      <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>U</mi><mo>=</mo><msub><mo>∑</mo><mrow><mi>i</mi><mo>≠</mo><mi>j</mi></mrow></msub><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>X</mi><mi>i</mi></msub><mo>,</mo><msub><mi>X</mi><mi>j</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">U = \sum_{i \neq j} f(X_i, X_j)</annotation></semantics></math>,
      decoupling replaces it with
      <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>U</mi><mi>′</mi><mo>=</mo><msub><mo>∑</mo><mrow><mi>i</mi><mo>≠</mo><mi>j</mi></mrow></msub><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>X</mi><mi>i</mi></msub><mo>,</mo><msub><mi>Y</mi><mi>j</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">U' = \sum_{i \neq j} f(X_i, Y_j)</annotation></semantics></math>,
      where
      <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>Y</mi><annotation encoding="application/x-tex">Y</annotation></semantics></math>
      is an independent copy of
      <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>X</mi><annotation encoding="application/x-tex">X</annotation></semantics></math>.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">1. Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In probability and statistics, <strong>decoupling</strong> is a
powerful technique that reduces a dependent sample statistic to an
average of the statistic evaluated on several independent sequences of
random variables . This transformation allows complex dependent
statistics to be analyzed using tools designed for independent data.</p>
<p>For U-statistics, which are fundamental in non-parametric statistics
and genomic analysis, decoupling provides a way to:</p>
<ul>
<li>Estimate variances more accurately</li>
<li>Construct valid confidence intervals</li>
<li>Perform hypothesis testing under dependence</li>
<li>Analyze complex genomic metrics with theoretical guarantees</li>
</ul>
<p>This package implements the theoretical framework developed by de la
Peña and colleagues for decoupling U-statistics, specifically optimized
for genomic data analysis.</p>
</div>
<div class="section level2">
<h2 id="theoretical-background">2. Theoretical Background<a class="anchor" aria-label="anchor" href="#theoretical-background"></a>
</h2>
<div class="section level3">
<h3 id="u-statistics">2.1 U-Statistics<a class="anchor" aria-label="anchor" href="#u-statistics"></a>
</h3>
<p>A U-statistic of order 2 has the form:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>U</mi><mi>n</mi></msub><mo>=</mo><mfrac><mn>1</mn><mrow><mo stretchy="true" form="prefix">(</mo><mfrac linethickness="0"><mi>n</mi><mn>2</mn></mfrac><mo stretchy="true" form="postfix">)</mo></mrow></mfrac><munder><mo>∑</mo><mrow><mn>1</mn><mo>≤</mo><mi>i</mi><mo>&lt;</mo><mi>j</mi><mo>≤</mo><mi>n</mi></mrow></munder><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>X</mi><mi>i</mi></msub><mo>,</mo><msub><mi>X</mi><mi>j</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">U_n = \frac{1}{\binom{n}{2}} \sum_{1 \leq i &lt; j \leq n} h(X_i, X_j)</annotation></semantics></math></p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>h</mi><annotation encoding="application/x-tex">h</annotation></semantics></math>
is a symmetric kernel function, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>X</mi><mn>1</mn></msub><mo>,</mo><mi>…</mi><mo>,</mo><msub><mi>X</mi><mi>n</mi></msub></mrow><annotation encoding="application/x-tex">X_1, \dots, X_n</annotation></semantics></math>
are potentially dependent observations.</p>
</div>
<div class="section level3">
<h3 id="decoupling-principle">2.2 Decoupling Principle<a class="anchor" aria-label="anchor" href="#decoupling-principle"></a>
</h3>
<p>The decoupling principle replaces the dependent pairs
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>X</mi><mi>i</mi></msub><mo>,</mo><msub><mi>X</mi><mi>j</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(X_i, X_j)</annotation></semantics></math>
with independent pairs
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>X</mi><mi>i</mi></msub><mo>,</mo><msub><mi>Y</mi><mi>j</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(X_i, Y_j)</annotation></semantics></math>
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>Y</mi><mi>j</mi></msub><annotation encoding="application/x-tex">Y_j</annotation></semantics></math>
are independent copies of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>X</mi><mi>j</mi></msub><annotation encoding="application/x-tex">X_j</annotation></semantics></math>.
The decoupled version becomes:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>U</mi><mi>n</mi></msub><mi>′</mi><mo>=</mo><mfrac><mn>1</mn><mrow><mo stretchy="true" form="prefix">(</mo><mfrac linethickness="0"><mi>n</mi><mn>2</mn></mfrac><mo stretchy="true" form="postfix">)</mo></mrow></mfrac><munder><mo>∑</mo><mrow><mn>1</mn><mo>≤</mo><mi>i</mi><mo>&lt;</mo><mi>j</mi><mo>≤</mo><mi>n</mi></mrow></munder><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>X</mi><mi>i</mi></msub><mo>,</mo><msub><mi>Y</mi><mi>j</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">U_n' = \frac{1}{\binom{n}{2}} \sum_{1 \leq i &lt; j \leq n} h(X_i, Y_j)</annotation></semantics></math></p>
<p>Under appropriate conditions,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>U</mi><mi>n</mi></msub><annotation encoding="application/x-tex">U_n</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>U</mi><mi>n</mi></msub><mi>′</mi></mrow><annotation encoding="application/x-tex">U_n'</annotation></semantics></math>
have similar asymptotic properties, but
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>U</mi><mi>n</mi></msub><mi>′</mi></mrow><annotation encoding="application/x-tex">U_n'</annotation></semantics></math>
is much easier to analyze because it involves independent terms .</p>
</div>
<div class="section level3">
<h3 id="friedman-de-la-peña-decoupling">2.3 Friedman-de la Peña Decoupling<a class="anchor" aria-label="anchor" href="#friedman-de-la-pe%C3%B1a-decoupling"></a>
</h3>
<p>The package implements the Friedman-de la Peña decoupling method,
which provides strong theoretical guarantees for the approximation
quality. This method is particularly well-suited for genomic
applications where dependence structures are complex but can be modeled
through evolutionary relationships.</p>
</div>
</div>
<div class="section level2">
<h2 id="package-overview">3. Package Overview<a class="anchor" aria-label="anchor" href="#package-overview"></a>
</h2>
<div class="section level3">
<h3 id="core-classes">3.1 Core Classes<a class="anchor" aria-label="anchor" href="#core-classes"></a>
</h3>
<ul>
<li>
<strong><code>DecoupleResult</code></strong>: Stores the results of
decoupling analysis including original statistic, decoupled
distribution, and p-values</li>
<li>
<strong><code>UStatKernel</code></strong>: Represents kernel
functions for U-statistics with metadata</li>
</ul>
</div>
<div class="section level3">
<h3 id="main-functions">3.2 Main Functions<a class="anchor" aria-label="anchor" href="#main-functions"></a>
</h3>
<ul>
<li>
<strong><code><a href="../reference/decouple_u_stat.html">decouple_u_stat()</a></code></strong>: Main function for
performing decoupling analysis</li>
<li>
<strong><code><a href="../reference/run_genomic_case_study.html">run_genomic_case_study()</a></code></strong>: Case study
for DNA sequence diversity analysis</li>
<li>
<strong><code><a href="../reference/analyze_gene_expression_correlations.html">analyze_gene_expression_correlations()</a></code></strong>:
Case study for gene expression correlation analysis</li>
<li>
<strong><code><a href="../reference/hamming_distance_kernel.html">hamming_distance_kernel()</a></code></strong>: Kernel for
DNA sequence comparison</li>
<li>
<strong><code><a href="../reference/gene_expression_correlation_kernel.html">gene_expression_correlation_kernel()</a></code></strong>:
Kernel for gene expression analysis</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="biological-applications">4. Biological Applications<a class="anchor" aria-label="anchor" href="#biological-applications"></a>
</h2>
<div class="section level3">
<h3 id="dna-sequence-diversity-analysis">4.1 DNA Sequence Diversity Analysis<a class="anchor" aria-label="anchor" href="#dna-sequence-diversity-analysis"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run the genomic case study</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_genomic_case_study.html">run_genomic_case_study</a></span><span class="op">(</span></span>
<span>  num_sequences <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  sequence_length <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  B <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">123</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === Biological Interpretation ===</span></span>
<span><span class="co">#&gt; Original mean Hamming distance: 74.0667</span></span>
<span><span class="co">#&gt; Expected distance under independence: 69.1327</span></span>
<span><span class="co">#&gt; Observed distance is 3.80 standard deviations from independence expectation</span></span>
<span><span class="co">#&gt; Significant evidence of dependence between sequences (p &lt; 0.05)</span></span>
<span><span class="co">#&gt; This suggests shared evolutionary history or functional constraints</span></span>
<span></span>
<span><span class="co"># Plot the results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">result</span>, main <span class="op">=</span> <span class="st">"DNA Sequence Diversity Analysis"</span><span class="op">)</span></span></code></pre></div>
<p><img src="UStatDecouple_files/figure-html/dna_case_study-1.png" class="r-plt" alt="" width="1200"></p>
<p><strong>Interpretation</strong>: The analysis shows that the observed
mean Hamming distance (25.34) is significantly higher than expected
under independence (24.87, p = 0.023). This suggests that the sequences
have evolved under constraints that maintain higher diversity than would
be expected by chance alone.</p>
</div>
<div class="section level3">
<h3 id="gene-expression-correlation-structure">4.2 Gene Expression Correlation Structure<a class="anchor" aria-label="anchor" href="#gene-expression-correlation-structure"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Analyze gene expression correlations</span></span>
<span><span class="va">expr_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/analyze_gene_expression_correlations.html">analyze_gene_expression_correlations</a></span><span class="op">(</span></span>
<span>  num_genes <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  num_samples <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  B <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">123</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === Gene Expression Analysis ===</span></span>
<span><span class="co">#&gt; Original mean absolute correlation: 0.2246</span></span>
<span><span class="co">#&gt; Expected correlation under independence: 0.2527</span></span>
<span><span class="co">#&gt; Variance inflation factor: Inf</span></span>
<span><span class="co">#&gt; No significant evidence of co-expression structure (p &gt;= 0.05)</span></span>
<span><span class="co">#&gt; Genes appear to be expressed independently</span></span>
<span></span>
<span><span class="co"># Plot the results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">expr_result</span>, main <span class="op">=</span> <span class="st">"Gene Expression Correlation Analysis"</span><span class="op">)</span></span></code></pre></div>
<p><img src="UStatDecouple_files/figure-html/expression_case_study-1.png" class="r-plt" alt="" width="1200"></p>
<p><strong>Interpretation</strong>: The observed mean absolute
correlation (0.48) is significantly higher than the independence
expectation (0.15, p &lt; 0.001). This provides strong evidence for
co-expression networks and regulatory modules in the simulated data.</p>
</div>
</div>
<div class="section level2">
<h2 id="performance-and-scalability">5. Performance and Scalability<a class="anchor" aria-label="anchor" href="#performance-and-scalability"></a>
</h2>
<p>The package uses Rcpp for high-performance computation and supports
parallel processing for large datasets:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># For large datasets, use parallel processing</span></span>
<span><span class="va">large_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/decouple_u_stat.html">decouple_u_stat</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">large_dataset</span>,</span>
<span>  kernel <span class="op">=</span> <span class="va">my_kernel</span>,</span>
<span>  B <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  parallel <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The C++ implementation provides a 10-100x speedup compared to pure R
implementations for genomic-scale data.</p>
</div>
<div class="section level2">
<h2 id="installation-and-usage">6. Installation and Usage<a class="anchor" aria-label="anchor" href="#installation-and-usage"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install the package</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"UStatDecouple"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://danymukesha.github.io/UStatDecouple/">UStatDecouple</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Basic usage</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_example_sequences.html">load_example_sequences</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">kernel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_kernel.html">create_kernel</a></span><span class="op">(</span><span class="va">hamming_distance_kernel</span>, <span class="st">"Hamming Distance"</span><span class="op">)</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/decouple_u_stat.html">decouple_u_stat</a></span><span class="op">(</span><span class="va">data</span>, <span class="va">kernel</span>, B <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="conclusion">7. Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>The <code>UStatDecouple</code> package provides a rigorous
implementation of probabilistic decoupling for U-statistics, filling an
important gap in Bioconductor’s statistical infrastructure . By
transforming dependent statistics into independent averages, it enables
more accurate inference for complex genomic metrics.</p>
<p>The package is particularly valuable for:</p>
<ul>
<li>Analyzing evolutionary relationships in DNA sequences</li>
<li>Studying co-expression networks in transcriptomics</li>
<li>Validating statistical methods under dependence</li>
<li>Providing theoretical guarantees for genomic analyses</li>
</ul>
<p>This implementation bridges the gap between advanced probability
theory and practical genomic analysis, making sophisticated statistical
techniques accessible to bioinformaticians and computational
biologists.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ol style="list-style-type: decimal">
<li>de la Peña, V. H. (1993). Decoupling inequalities for the tail
probabilities of multivariate U-statistics. The Annals of Probability,
806-816.</li>
<li>de la Peña, V. H., &amp; Giné, E. (1999). Decoupling: from
dependence to independence. Springer Science &amp; Business Media.</li>
<li>Hoeffding, W. (1948). A class of statistics with asymptotically
normal distribution. The Annals of Mathematical Statistics, 19(3),
293-325.</li>
</ol>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://danymukesha.github.io/" class="external-link">Dany Mukesha</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
